src/AdobeAnalyticsHelper.php

```diff
@@ -185,73 +192,165 @@ public function getVariables() {
    *   Build array.
    */
   public function renderMarkup() {
-    if ($this->skipTracking()) {
-      return [];
+
+    $build = [];
+    $build['#theme'] = 'analytics_code';
+
+    if ($this->cdnConfig->get('mode') == 'cdn') {
+      $build = $this->renderCdnMarkup($build);
+    }
+
+    if ($this->datalayerConfig->get('data_layer_enabled') == 1) {
+      $build = $this->renderDatalayerMarkup($build);
     }
 
-    // Extract module settings.
-    $js_file_location = $this->config->get('js_file_location');
-    $codesnippet = $this->config->get('codesnippet');
-    $version = $this->config->get("version");
-    $nojs = !empty($this->config->get("image_file_location")) ? $this->config->get("image_file_location") : NULL;
+    if ($this->cdnConfig->get('mode') == 'general') {
+
+      if ($this->skipTracking(1)) {
+        return [];
+      }
+
+      // Extract module settings.
+      $js_file_location = $this->cdnConfig->get('js_file_location');
+      $codesnippet = $this->cdnConfig->get('codesnippet');
+      $version = $this->cdnConfig->get("version");
+      $nojs = !empty($this->cdnConfig->get("image_file_location")) ? $this->cdnConfig->get("image_file_location") : NULL;
 
+      // Extract entity overrides.
+      list ($include_main_codesnippet, $include_custom_variables, $entity_snippet) = $this->extractEntityOverrides();
 
-    // Extract entity overrides.
-    list ($include_main_codesnippet, $include_custom_variables, $entity_snippet) = $this->extractEntityOverrides();
+      // Format and combine variables in the "right" order
+      // Right order is the code file (list likely to be maintained)
+      // Then admin settings with codesnippet first and finally taxonomy->vars.
```

Comment:
Each line doesn't need a capital letter. Make sure each line is as close to (but no more than) 80 characters.